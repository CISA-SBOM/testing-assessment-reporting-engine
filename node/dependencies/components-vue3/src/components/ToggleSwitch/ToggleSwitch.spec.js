import { describe, expect, it } from 'vitest'
import { shallowMount } from '@vue/test-utils'
import Component from "./ToggleSwitch.vue";

describe("ToggleSwitch.vue", () => {
  it("is a Vue instance", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.vm).toBeTruthy();
  });
  it("matches snapshot", () => {
    const wrapper = shallowMount(Component);
    expect(wrapper.element).toMatchSnapshot();
  });
  it("matches snapshot with no props assigned", () => {
    const propsData = {};
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("matches snapshot with prop assigned", () => {
    const propsData = {
      disabled: false,
      variant: 'primary',
      value: false
    };
    const wrapper = shallowMount(Component, { propsData });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
