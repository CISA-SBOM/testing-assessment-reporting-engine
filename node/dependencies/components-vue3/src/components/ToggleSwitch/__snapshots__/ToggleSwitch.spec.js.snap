// Vitest Snapshot v1

exports[`ToggleSwitch.vue > matches snapshot 1`] = `
<div
  class="flex items-center"
  data-id="sds-toggle-switch"
  data-v-faa1240c=""
>
  <button
    aria-checked="false"
    class="bg-gray-700 disabled:bg-opacity-50 hover:shadow-md relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none disabled:bg-gray-700 disabled:bg-opacity-50 disabled:shadow-none"
    data-v-faa1240c=""
    role="switch"
    type="button"
  >
    <span
      class="translate-x-0 toggle-off-shadow pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow-lg transform transition ease-in-out duration-200"
      data-v-faa1240c=""
    />
    <svg
      class="translate-x-0 block pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-gray-700"
      data-v-faa1240c=""
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1490 1322q0 40-28 68l-136 136q-28 28-68 28t-68-28l-294-294-294 294q-28 28-68 28t-68-28l-136-136q-28-28-28-68t28-68l294-294-294-294q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 294 294-294q28-28 68-28t68 28l136 136q28 28 28 68t-28 68l-294 294 294 294q28 28 28 68z"
        data-v-faa1240c=""
      />
    </svg>
    <span
      class="ml-3"
      data-v-faa1240c=""
    />
    <svg
      class="translate-x-0 hidden pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-blue-500"
      data-v-faa1240c=""
      viewBox="0 0 1792 1792"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1671 566q0 40-28 68l-724 724-136 136q-28 28-68 28t-68-28l-136-136-362-362q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 295 656-657q28-28 68-28t68 28l136 136q28 28 28 68z"
        data-v-faa1240c=""
      />
    </svg>
  </button>
</div>
`;

exports[`ToggleSwitch.vue > matches snapshot with no props assigned 1`] = `
"<div data-id=\\"sds-toggle-switch\\" class=\\"flex items-center\\" data-v-faa1240c=\\"\\"><button type=\\"button\\" class=\\"bg-gray-700 disabled:bg-opacity-50 hover:shadow-md relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none disabled:bg-gray-700 disabled:bg-opacity-50 disabled:shadow-none\\" role=\\"switch\\" aria-checked=\\"false\\" data-v-faa1240c=\\"\\"><span class=\\"translate-x-0 toggle-off-shadow pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow-lg transform transition ease-in-out duration-200\\" data-v-faa1240c=\\"\\"></span><svg class=\\"translate-x-0 block pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-gray-700\\" viewBox=\\"0 0 1792 1792\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-faa1240c=\\"\\">
      <path d=\\"M1490 1322q0 40-28 68l-136 136q-28 28-68 28t-68-28l-294-294-294 294q-28 28-68 28t-68-28l-136-136q-28-28-28-68t28-68l294-294-294-294q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 294 294-294q28-28 68-28t68 28l136 136q28 28 28 68t-28 68l-294 294 294 294q28 28 28 68z\\" data-v-faa1240c=\\"\\"></path>
    </svg><span class=\\"ml-3\\" data-v-faa1240c=\\"\\"></span><svg class=\\"translate-x-0 hidden pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-blue-500\\" viewBox=\\"0 0 1792 1792\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-faa1240c=\\"\\">
      <path d=\\"M1671 566q0 40-28 68l-724 724-136 136q-28 28-68 28t-68-28l-136-136-362-362q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 295 656-657q28-28 68-28t68 28l136 136q28 28 28 68z\\" data-v-faa1240c=\\"\\"></path>
    </svg></button></div>"
`;

exports[`ToggleSwitch.vue > matches snapshot with prop assigned 1`] = `
"<div data-id=\\"sds-toggle-switch\\" class=\\"flex items-center\\" value=\\"false\\" data-v-faa1240c=\\"\\"><button type=\\"button\\" class=\\"bg-gray-700 disabled:bg-opacity-50 hover:shadow-md relative inline-flex flex-shrink-0 h-6 w-11 border-2 border-transparent rounded-full cursor-pointer transition-colors ease-in-out duration-200 focus:outline-none disabled:bg-gray-700 disabled:bg-opacity-50 disabled:shadow-none\\" role=\\"switch\\" aria-checked=\\"false\\" data-v-faa1240c=\\"\\"><span class=\\"translate-x-0 toggle-off-shadow pointer-events-none inline-block h-5 w-5 rounded-full bg-white shadow-lg transform transition ease-in-out duration-200\\" data-v-faa1240c=\\"\\"></span><svg class=\\"translate-x-0 block pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-gray-700\\" viewBox=\\"0 0 1792 1792\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-faa1240c=\\"\\">
      <path d=\\"M1490 1322q0 40-28 68l-136 136q-28 28-68 28t-68-28l-294-294-294 294q-28 28-68 28t-68-28l-136-136q-28-28-28-68t28-68l294-294-294-294q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 294 294-294q28-28 68-28t68 28l136 136q28 28 28 68t-28 68l-294 294 294 294q28 28 28 68z\\" data-v-faa1240c=\\"\\"></path>
    </svg><span class=\\"ml-3\\" data-v-faa1240c=\\"\\"></span><svg class=\\"translate-x-0 hidden pointer-events-none absolute h-4 w-4 mt-0.5 ml-0.5 transform ring-0 transition ease-in-out duration-400 fill-current text-blue-500\\" viewBox=\\"0 0 1792 1792\\" xmlns=\\"http://www.w3.org/2000/svg\\" data-v-faa1240c=\\"\\">
      <path d=\\"M1671 566q0 40-28 68l-724 724-136 136q-28 28-68 28t-68-28l-136-136-362-362q-28-28-28-68t28-68l136-136q28-28 68-28t68 28l294 295 656-657q28-28 68-28t68 28l136 136q28 28 28 68z\\" data-v-faa1240c=\\"\\"></path>
    </svg></button></div>"
`;
